server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java

```diff
@@ -207,6 +208,16 @@ public final class IndexScopedSettings extends AbstractScopedSettings {
         )
     );
 
+    /**
+     * Map of feature flag name to feature-flagged index setting. Once each feature
+     * is ready for production release, the feature flag can be removed, and the
+     * setting should be moved to {@link #BUILT_IN_INDEX_SETTINGS}.
+     */
+    public static final Map<String, Setting> FEATURE_FLAGGED_INDEX_SETTINGS = Map.of(
```

Comment:
Note - `Map.of` is only available from Java 9+

Reply:
This shouldn't be an issue since we have to remain 11+ due to lucene 9.